### DISCUSSION TITLE
prevent hiding a map if it's the only map left unhidden in a mapset

### DISCUSSION THREAD

**Discussion by iSimpFor** [ppy/osu#16855] [Ideas]
when you hide a map and it's the last map in a mapset(in all rulesets), the whole mapset disappears(and is inaccessible) until you hit the "restore all hidden maps button"

in cases where people hide a LOT of maps(for one reason or another), rehiding all maps that you didn't want to be unhidden becomes a time-consuming and annoying task

preventing the last map of a mapset across all rulesets to be hidden avoids the situation where a mapset becomes inaccessible

**Comment by bdach** [comment]
but why would you hide the map in the first place?

also it sounds like misuse of the hide functionality. to get better management options you probably want to use collections instead.

**Reply by iSimpFor** [reply]
in said case, i would say that someone could do this mistakenly when they're doing something with the hide function(such as hiding other maps)

**Reply by bdach** [reply]
that argument also works if you mistakenly hide a difficulty that isn't the last one in a set.

so the solution if anything would be to allow the user to see and unhide mistakenly hidden difficulties, in general, rather than blocking the hide action in overly specific circumstances.

**Reply by iSimpFor** [reply]
that should be fine but how would this be implemented?

**Reply by frenzibyte** [reply]
Can imagine it being implemented very easily by a "Show hidden beatmaps" setting in either song select header or in game settings which the `CarouselBeatmap` would respond to and update their `Alpha`/presence state accordingly. Not sure why you're asking that.
